internal/sdkprovider/service/vpc/vpc_peering_connection_state.go

```diff
@@ -28,7 +28,7 @@ func getDiagnosticsFromState(pc peeringConnectionState) diag.Diagnostics {
 					"completes the setup in their cloud account. The steps needed in the user cloud "+
 					"account depend on the used cloud provider. Find more in the state info: %s",
 					stateInfoToString(pc.GetStateInfo()))}}
-		case "DELETED":
+		case "DELETED", "DELETING":
```

Comment:
nit: I would replace literals with the constants from the client https://github.com/aiven/go-client-codegen/blob/main/handler/organizationvpc/organizationvpc.go
That would help to:
1. prove the values are valid, and do not have typos
2. verify breaking changes

Reply:
We use strings for the state all over these resources.
1. Changing it here wouldn't ,make any difference
2. This string also works with the types coming from the old client. Comparing them would anyway require converting them to string.

Reply:
> We use strings for the state all over these resources.

Which resources are you referring to? IMO we should migrate to codegen client enums where possible. The states may change on the backend and using codegen client makes it clearer where the "magic" string comes from and what it relates to.

> This string also works with the types coming from the old client. Comparing them would anyway require converting them to string.

Are we using both clients here? If so, as the old one is deprecated, shouldn't we migrate to the codegen client? That would make this point moot. Additionally, we seem to be missing a few connection states here, namely "APPROVED", "APPROVED_PEER_REQUESTED" and "ERROR".

Reply:
This code is a copy from old peering connections which use the old client